### DISCUSSION TITLE
how to explicitly link the absl_log_flags.lib

### DISCUSSION THREAD

**Discussion by wony-zheng** [abseil/abseil-cpp#1439] [Q&A]
as absl_log_flags.lib export nothing in flags.h. when link in vs, the lib seem to skipped in linking. when compiled exe run with --helpfull param, there is no log flag parameter displayed. how can i force/explicitly link the absl_log_flags.ib in visual studio?

**Comment by univ2218** [comment]
Looking at the abseil source code, there is the following comment:

> Warning: Many linkers will strip the contents of this library because its
> symbols are only used in a global constructor. A workaround is for clients
> to link this using $<LINK_LIBRARY:WHOLE_ARCHIVE,absl::log_flags> instead of
> the plain absl::log_flags.
> TODO(b/320467376): Implement the equivalent of Bazel's alwayslink=True.

I was therefore able to link to the absl::log_flags library with cmake by replacing the CMakeLists.txt line:

`target_link_libraries(myTarget PUBLIC absl::log_flags)`

with

`target_link_libraries(myTarget PUBLIC "$<LINK_LIBRARY:WHOLE_ARCHIVE,absl::log_flags>")`
